### PR TITLE
[WB-6234]: Launch metadata in file

### PR DESCRIPTION
**Pull request by KyleGoyette**
https://wandb.atlassian.net/browse/WB-6234

Description
-----------

Puts the launch metadata (launch spec, dockerfile and docker args) into a file and uploads it to the server for all runs using launch.

Testing
-------

How was this PR tested?

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------
- Store launch metadata in file

------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/sdk/launch/docker.py [hunk 1]
```diff
@@ -159,11 +162,15 @@ def build_docker_image(
         workdir_line=workdir_line,
         name_line=name_line,
     )
+
+    launch_metadata = create_launch_metadata(launch_project, docker_args, dockerfile)
```
Comment (vanpelt) [inline wandb/sdk/launch/docker.py:166]: Is dockerfile a path or the actual contents?  It looks like we're setting the variable to a path on line 178 below...

Comment (KyleGoyette) [reply wandb/sdk/launch/docker.py:166]: The actual contents, bad name, will fix that

Comment (KyleGoyette) [reply wandb/sdk/launch/docker.py:166]: I'm also wondering, if I should refactor this to also include the actual docker command (with API key hidden of course)

Comment (vanpelt) [reply wandb/sdk/launch/docker.py:166]: Yeah, sounds like a good idea.  I suppose a user could have other sensitive information in the command...  I suppose at some point in the future we could give users a way to mark certain parameters as sensitive.

**Review by stephchen — COMMENTED** [review]
where does the dockerfile get written to the config file? also what is the current specification for what is contained at DEFAULT_CONFIG_PATH — i think it has changed drastically from when i originally wrote that (i think with the intention of saving the user's uploaded json config somewhere?) and i'm guessing the pathname of `launch_override_config.json` is not accurate anymore

**Comment by KyleGoyette** [comment]
Good points. Now the DEFAULT_CONFIG_PATH will be the DEFAULT_LAUNCH_METADATA_PATH. I've changed the file name to launch_metadata.json. Now it just contains the launch_spec, along with the command used to run the container and the docker file contents.

**Pull request merged**